### ISSUE TITLE
Ruse is Legion

### ISSUE THREAD

**Issue by bluegrouse92-lgtm** [N-A-pixel/Ruse-Public#9]
# Ruse is Legion

## Research vision

Ruse is not being studied as a single ever-larger language model. The experimental direction is a **society of models**: independently trained Ruse Native lineages, including potentially very small specialist models, evaluated and routed according to measured capability.

The research question is whether a heterogeneous population of models can collectively provide broader validated capability than any single small checkpoint while preserving complete lineage and evaluation evidence.

> **Ruse is not one brain. Ruse is the institution.**

## Core principle

**EVERYONE GETS A CHANCE.  
CITIZENSHIP IS EARNED.  
EVERY CITIZEN GETS A JOB.**

Experimental checkpoints are not automatically citizens. Models pass through controlled education and evaluation. Failed experiments can remain preserved as provenance without every failed weight set becoming an operational worker.

Once a model earns citizenship, the goal is not to discard it merely because a newer or larger model appears. Roles should emerge from **measured capability**.

**Preservation does not mean promotion.**

## Play to strengths

Possible research roles include:

- intent classification
- language identification
- routing
- retrieval decisions
- formatting / structured output
- JSON transformation
- summarisation
- arithmetic checking
- code/syntax checking
- translation
- domain-specific knowledge work
- consistency checking
- planning
- general language work

These are examples, not predetermined careers.

## Dynamic hierarchy

The hierarchy represents operational responsibility rather than a permanent ranking of intelligence.

A capability registry should track:

- citizen identity and lineage
- architecture / parameter count
- tokenizer
- developmental age
- education history
- canonical checkpoint
- evaluation history
- demonstrated strengths
- demonstrated weaknesses
- specialist roles
- compute/runtime cost
- latency
- reliability
- routing eligibility
- dependencies / supervisors
- current operational position

Model selection should be evidence-based rather than defaulting to the largest available checkpoint.

## Gary

Gary — **RN-50M-32K-ML** — occupies a special architectural root position in the experimental hierarchy.

This does **not** mean Gary is assumed to be the most capable model at every task. Operational work remains dynamic and delegated according to measured strengths.

Gary's development helped force stronger curriculum interleaving, contamination auditing, tokenizer-aware evaluation, repetition measurement and preflight controls.

## First cohort

The first experimental cohort reached the common **E3 developmental rung (~5M cumulative education tokens)**. The separate post-E3 English intervention (E3-EI) was also completed, with pristine E3 states preserved as graduation baselines.

Core lineages:

- RN-20M Clone
- corrected RN-50M
- RN-100M
- Gary / RN-50M-32K-ML

The V1.1 parameter-scaling family and Gary's multilingual 32K lineage remain experimentally distinct.

## Capability-based routing experiment

The planned orchestration research separates:

**Identity** — which model lineage is being used.

**Capability** — what evaluation evidence says it can currently do.

**Role** — what bounded work it is assigned.

**Authority** — where it sits in the experimental governance hierarchy.

**Routing** — when it should actually be invoked.

A task may involve one model or several models in a planner/worker/verifier arrangement. The purpose of this research is to test whether smaller qualified specialists can handle bounded work instead of always invoking a larger model.

## Tiny specialists

A major part of the experiment is deliberately allowing **tiny models** to provide narrow pieces of broader system capability.

A future 8M, 10M or 20M-class model might exist because it performs one bounded operation extremely well and cheaply. A hypothetical tiny model that only produces highly reliable JSON is not a failed generalist; it is a potential specialist if the evaluation supports that role.

## Ruse School

As the population grows, long controlled education runs move into **Ruse School**, a remote reproducible training environment.

A citizen package includes:

- canonical checkpoint
- optimizer state where continuity applies
- tokenizer identity
- lineage identity
- exact developmental age
- education ledger
- hashes and provenance

The School pipeline covers corpus selection, curriculum construction, tokenization, deduplication, novelty analysis, contamination checks, exact token-budget accounting, optimizer continuity, unattended training, recovery checkpoints, frozen evaluation, checkpoint reload verification, hashing and provenance recording.

In conventional MLOps terms, Ruse School is a **remote training control plane plus reproducible data/training pipeline** for a population of persistent model lineages.

Terminology maps to concrete technical concepts:

- **citizen** = accepted persistent model lineage with canonical identity;
- **age** = cumulative accounted pretraining-token exposure;
- **school year / E-stage** = predefined developmental training boundary;
- **lesson** = bounded provenance-tracked curriculum/data slice;
- **school** = remote corpus, curriculum, training, evaluation and artifact-management environment;
- **report card / exam** = frozen evaluation suite and metrics;
- **graduation package** = immutable checkpoint/optimizer/tokenizer/manifest/evaluation artifact set;
- **job** = capability-registry-backed routing role;
- **Legion** = multi-model orchestration research architecture.

The lifecycle is:

```text
package citizen + education state
→ transfer
→ verify hashes before training
→ admit to Ruse School
→ construct and audit authorized curriculum
→ train to exact developmental boundary
→ freeze / reload / hash
→ run required evaluation
→ produce graduate package
→ register canonical graduate
```

Changing compute location does not create a new citizen or lineage.

## Post-E3 English intervention

E3-EI consumed 100,096 model tokens for each V1.1 citizen and 100,736 for Gary due to his separate tokenizer.

The intervention shifted the attractor landscape rather than simply improving or degrading the same pattern. Encyclopedic/scientific/geographic loops were displaced by narrative/literary loops. This demonstrated strong sensitivity to concentrated recent education while showing that attractor pathology remained.

## Quadruplet runner cohort

The tiny-specialist concept is represented by **RN-8M-Q1/Q2/Q3/Q4**, each 8,523,008 parameters.

The four receive synchronized common education while retaining independent weights and optimizer trajectories. At controlled barriers, each receives frozen textual material produced by the other siblings; no weight, gradient, hidden-state or optimizer sharing is allowed.

The cohort exists to measure useful divergence and later identify evidence-backed bounded roles.

```text
COMMON EDUCATION
→ CONTROLLED SIBLING AWARENESS
→ FREEZE FOUR GENERAL GRADUATES
→ CHARACTERISE DIFFERENCES
→ IDENTIFY CANDIDATE ROLES
→ separately authorize specialist education
→ qualify specialist descendants
```

The objective is not to force every citizen toward the same endpoint. The objective is to study **different kinds of competence** under reproducible education and evidence-backed routing.

**RUSE IS LEGION.**

**Comment by bluegrouse92-lgtm** [comment]
## Post-E3 population and orchestration plan

After the first E3 cohort, the research programme expands to a heterogeneous population rather than collapsing back to one larger model.

Planned/realized directions include preserving canonical E3 states, resuming OG RN-20M, creating the RN-8M cohort, adding independent ~100M and ~300M lineages, and using capability/evaluation records to study evidence-based routing between preserved models.

The orchestration research requires lineage, checkpoint, education, capability and role registries plus explicit School admission/graduation boundaries. Compute location does not change model identity.

**Comment by bluegrouse92-lgtm** [comment]
## E3-EI — bounded intervention preserved, not promoted

The post-E3 English intervention produced strong behavioral redistribution across the four E3 lineages but did not reliably improve semantic control. Historical validation and holdout losses worsened, while generation attractors shifted rather than disappearing.

The experimental descendants are frozen for research value. Canonical E3 remains the main-line historical boundary.

**Comment by bluegrouse92-lgtm** [comment]
## 23 September 2026 — planned population becomes real

The post-E3 expansion moved from plan to implementation:

- OG RN-20M resumed from its protected historical lineage;
- RN-8M-Q1–Q4 created as four independent 8,523,008-parameter siblings;
- RN-100M-new created as a fresh independent ~100M observation;
- RN-300M-new-20260923 created and exercised through the Oracle package/transfer/integrity boundary.

The population is now heterogeneous enough to support concrete experiments in lineage-aware, evidence-based multi-model routing.

**Comment by bluegrouse92-lgtm** [comment]
## Ruse School Phase 2 — pre-education pipeline operational

RN-300M completed the first end-to-end pre-education School path without any weight mutation: package integrity, canonical ownership transfer, proof-library admission, tokenizer verification, novelty/contamination checks, frozen lesson construction and training-lock refusal were all exercised successfully.

This established the separation between preparing education and authorizing training.

**Comment by bluegrouse92-lgtm** [comment]
## 24 September 2026 — Legion population update

RN-8M Q1/Q2/Q3/Q4 entered a new common-classroom stage from their verified 400K descendants. Each is authorized for exactly **+5,000,000 optimized targets**, with independent weights and synchronized barriered peer-review exposure.

RN-300M's E0 and evaluation are complete; no RN-300M process is currently running.

The architectural principle remains: separate model identities, separate provenance and separate educational state, with later role assignment based on evidence rather than model size or naming.
